### PR TITLE
NMS-15296: Update Fix Karaf script instruction in Housekeeping Tasks

### PR DESCRIPTION
**Pull request by emarsh1995**
- Update `operation/deep-dive/admin/housekeeping.adoc`

### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here, or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready. Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15296

### CODE REVIEW

File: docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc [hunk 1]
```diff
@@ -27,20 +27,17 @@ SELECT eventid FROM notifications WHERE eventid = events.eventid)
...
-NOTE: Back up `/opt/opennms/etc` before running the script.
+If your {page-component-title} instance fails to start after upgrading, you can run the Fix Karaf script to delete cached items.
+The script tries to fix configuration problems by pruning `/opt/opennms/data` and restoring all Karaf-related configuration files to a pristine state.
 
 [source, console]
-----
 /opt/opennms/bin/fix-karaf-setup.sh
-----
+
+IMPORTANT: The Fix Karaf script can be destructive to files that may contain user-specific configuration settings.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc:40]:
```suggestion
WARNING: The Fix Karaf script can be destructive to files that may contain user-specific configuration settings.
```
I think this is a bit stronger than Important, and might draw the reader's attention better.

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
just one change. Thanks!

**Review by Bonrob2 — COMMENTED** [review]
LGTM! Thank you!

File: docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 
 [[prune-events]]
 === Prune unneeded events
+
 Use vacuumd or cron to regularly prune unneeded events.
 This helps to clean up your events table, and reduces the amount of data in your backups.
 We also recommend doing this before you begin the {page-component-title} upgrade process.
 
-For example, run the following to delete events older than six weeks that have no associated outages:
+For example, run the following code to delete events older than six weeks that have no associated outages:
```
Comment (mmahacek) [inline docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc:16]: The following is part of the default vacuumd config, and is automatically run every 24hrs the server is up. It shouldn't need to be run manually.

**Review by Bonrob2 — APPROVED** [review]
LGTM!

File: docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc [hunk 3]
```diff
@@ -1,46 +1,29 @@
...
-(SELECT svclosteventid FROM outages WHERE svclosteventid = events.eventid
-UNION
-SELECT svcregainedeventid FROM outages WHERE svcregainedeventid = events.eventid
-UNION
-SELECT eventid FROM notifications WHERE eventid = events.eventid)
-AND eventtime < now() - interval '6 weeks';
-----
-
-== After Upgrading
+== After upgrading
```
Comment (mmahacek) [inline docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc:16]: This is an empty header with no content.

Comment (Bonrob2) [reply docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc:16]: It's the same as the "Prior to upgrading" heading. It is unusual that there is only one item under each, but it's a way to separate the pre- and post- upgrading tasks visually.

**Review by Bonrob2 — APPROVED** [review]
LGTM!

File: docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc [hunk 4]
```diff
@@ -1,46 +1,29 @@
...
-NOTE: Back up `/opt/opennms/etc` before running the script.
+If your {page-component-title} instance fails to start after upgrading, you can run the Fix Karaf script to delete cached items.
+The script tries to fix configuration problems by pruning `/opt/opennms/data` and restoring all Karaf-related configuration files to a pristine state.
 
 [source, console]
-----
 /opt/opennms/bin/fix-karaf-setup.sh
-----
+
+WARNING: The Fix Karaf script can be destructive to files that may contain user-specific configuration settings.
```
Comment (deejgregor) [inline docs/modules/operation/pages/deep-dive/admin/housekeeping/introduction.adoc:26]: This is where you might want to move the note that was removed:
```
NOTE: Back up `/opt/opennms/etc` before running the script.
```

**Review by deejgregor — DISMISSED** [review]
I made one suggestion, but it is non-blocking. The fix-karaf-setup.sh changes seem good to me.

I don't have much insight into the deletion of the SQL example for deleting events.

**Review by deejgregor — APPROVED** [review]

File: docs/modules/operation/pages/deep-dive/admin/housekeeping.adoc [hunk 5]
```diff
@@ -28,15 +29,13 @@ AND eventtime < now() - interval '6 weeks';
...
+If your {page-component-title} instance fails to start after upgrading, you can run the Fix Karaf script to delete cached items.
+The script tries to fix configuration problems by pruning `/opt/opennms/data` and restoring all Karaf-related configuration files to a pristine state.
 
 [source, console]
-----
 /opt/opennms/bin/fix-karaf-setup.sh
-----
+
+WARNING: The Fix Karaf script can be destructive to files that may contain user-specific configuration settings.
+It is recommended that you back up `/opt/opennms/etc` before running the script.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/admin/housekeeping.adoc:41]:
```suggestion
We recommend that you back up `/opt/opennms/etc` before running the script.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
one nit!

**Review by Bonrob2 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**